lib/Coverage.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {helper} = require('./helper');
+
+class Coverage {
+  /**
+   * @param {!Puppeteer.Session} client
+   */
+  constructor(client) {
+    this._client = client;
+    this._jsCoverage = false;
+    this._scripts = new Map();
+    this._jsEventListeners = [];
+  }
+
+  async startJSCoverage() {
+    console.assert(!this._jsCoverage, 'JSCoverage is already enabled');
+    this._jsCoverage = true;
+    this._scripts.clear();
+    this._jsEventListeners = [
+      helper.addEventListener(this._client, 'Debugger.scriptParsed', this._onScriptParsed.bind(this)),
+      helper.addEventListener(this._client, 'Runtime.executionContextsCleared', this._onExecutionContextsCleared.bind(this))
+    ];
+    await Promise.all([
+      this._client.send('Profiler.enable'),
```

Comment:
Is the order of these methods important? Are we relying on them being synchronous on the backend?

Reply:
yes: we should enable profiler before starting coverage, and similarly we're enabling debugger before skipping pauses.

Reply:
Will this conflict if we use profiler for something else later?